ext/openssl/openssl.stub.php

```diff
@@ -95,105 +92,100 @@ function openssl_pkey_free(OpenSSLAsymmetricKey $key): void {}
 
-/** @param OpenSSLCertificate|array|string $recipcerts */
-function openssl_cms_encrypt(string $infile, string $outfile, $recipcerts, ?array $headers, int $flags = 0, int $encoding = OPENSSL_ENCODING_SMIME,  int $cipher = OPENSSL_CIPHER_RC2_40): bool {}
+/** @param OpenSSLCertificate|array|string $certificate */
+function openssl_cms_encrypt(string $filename, string $output_filename, $certificate, ?array $headers, int $flags = 0, int $encoding = OPENSSL_ENCODING_SMIME,  int $cipher_algorithm = OPENSSL_CIPHER_RC2_40): bool {}
 
-/** @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $signkey */
-function openssl_cms_sign(string $infile, string $outfile, OpenSSLCertificate|string $signcert, $signkey, ?array $headers, int $flags = 0, int $encoding = OPENSSL_ENCODING_SMIME, ?string $extracertsfilename = null): bool {}
+/** @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key */
+function openssl_cms_sign(string $filename, string $output_filename, OpenSSLCertificate|string $certificate, $private_key, ?array $headers, int $flags = 0, int $encoding = OPENSSL_ENCODING_SMIME, ?string $untrusted_certificates_filename = null): bool {}
 
 /**
- * @param OpenSSLCertificate|string $recipcert
- * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $recipkey
+ * @param OpenSSLCertificate|string $certificate
+ * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string|null $private_key
  */
-function openssl_cms_decrypt(string $infilename, string $outfilename, $recipcert, $recipkey = UNKNOWN, int $encoding = OPENSSL_ENCODING_SMIME): bool {}
+function openssl_cms_decrypt(string $filename, string $output_filename, $certificate, $private_key = null, int $encoding = OPENSSL_ENCODING_SMIME): bool {}
 
-/** @param array $certs */
-function openssl_cms_read(string $infilename, &$certs): bool {}
+/** @param array $certificates */
+function openssl_cms_read(string $filename, &$certificates): bool {}
 
 /**
- * @param string $crypted
- * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $key
+ * @param string $encrypted_data
+ * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key
  */
-function openssl_private_encrypt(string $data, &$crypted, $key, int $padding = OPENSSL_PKCS1_PADDING): bool {}
+function openssl_private_encrypt(string $data, &$encrypted_data, $private_key, int $padding = OPENSSL_PKCS1_PADDING): bool {}
 
 /**
- * @param string $crypted
- * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $key
+ * @param string $encrypted_data
+ * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key
  */
-function openssl_private_decrypt(string $data, &$crypted, $key, int $padding = OPENSSL_PKCS1_PADDING): bool {}
+function openssl_private_decrypt(string $data, &$encrypted_data, $private_key, int $padding = OPENSSL_PKCS1_PADDING): bool {}
```

Comment:
This is really other way round. First param is `from` (encrypted data) and the second param is `to` which is decrypted data. If you want to keep first as data, then the second one should be `decrypted_data`.

Reply:
Oops, it's very fortunate that you had another look :) Thank you!